app/lib/actions/pulp/consumer/sync_node.rb

```diff
@@ -9,6 +9,29 @@ class SyncNode < AbstractSyncNodeTask
         end
 
         def invoke_external_task
+          task = invoke_external_task_based_on_repos_id
+          schedule_timeout(Setting['content_action_accept_timeout'])
+          task
+        end
+
+        def process_timeout
+          accept_timeout = Setting['content_action_accept_timeout']
+          finish_timeout = Setting['content_action_finish_timeout']
+
+          if pulp_state == 'waiting'
+            fail _("Host did not respond within %s seconds. Is katello-agent installed and goferd running on the Host?") % accept_timeout
+          elsif pulp_state == 'unknown'
+            fail _("Unknown Status during sync. Is katello-agent installed and goferd running on the Host?")
+          else
+            if output[:sync_task_is_accepted].nil?
+              output[:sync_task_is_accepted] ||= true
+            else
+              fail _("Host did not finish sync within %s seconds. Is katello-agent installed and goferd running on the Host?") % finish_timeout
+            end
```

Comment:
I don't actually see you re-scheduling the timeout for the 'finish_timeout', so I'm not sure how it would ever hit this else.